sentry/src/main/java/io/sentry/instrumentation/file/SentryFileInputStream.java

```diff
@@ -0,0 +1,144 @@
+package io.sentry.instrumentation.file;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.HubAdapter;
+import io.sentry.IHub;
+import io.sentry.ISpan;
+import java.io.File;
+import java.io.FileDescriptor;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * An implementation of {@link java.io.FileInputStream} that creates a {@link io.sentry.ISpan} for
+ * reading operation with filename and byte count set as description
+ *
+ * <p>Note, that span is started when this InputStream is instantiated via constructor and finishes
+ * when the {@link java.io.FileInputStream#close()} is called.
+ */
+@Open
```

Comment:
if we are not extending, should we make it `final`?

Reply:
I thought it's good to leave it open, because the users might want to extend it, for instance, if they have their own impls of FIOS, but also want to have them instrumented

Reply:
mm, yeah, wondering if it makes sense to keep it closed, easier to debug new issues, and later open it if people need it, its always easier to open, always harder to close it.

Reply:
Actually, good point. We have a Factory which they still can use to pass their impls, and we will do wrapping and instrumenting